### PR TITLE
Add ignore goroutine return values tenet

### PR DESCRIPTION
**Pull request by CallanTaylor**

### CODE REVIEW

File: tenets/codelingo/go/ignore-goroutine-return-values/codelingo.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      go.go_stmt(depth = any):
+        go.call_expr:
+          any_of:
+            go.func_lit:
+              go.func_type:
+                go.field_list
+                go.field_list
+            edge("calls"):
+              go.func_decl:
+                go.return_stmt(depth = any)
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/ignore-goroutine-return-values/codelingo.yaml]: I'm guessing this means that if you have a `return` in a `func_decl` we assume that it returns a value? You can return without returning a value. [pg link for](https://play.golang.org/p/gf5aXD7zjY5):

```go
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Hello, playground")
	if true {
		return	
	}
	fmt.Println("Hello, playground")
}
```

Probably safer to use that same pattern as above:

```yaml
go.func_decl:
  go.func_type:
    go.field_list
    go.field_list
```

File: tenets/codelingo/go/ignore-goroutine-return-values/example.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+func main() {
+	go A() // Issue
+        go B() // Non Issue
+	<-time.After(time.Second)
+
+	go func() string {
+	    word := "Hello"
+	    return word // Issue
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/ignore-goroutine-return-values/example.go]:
```suggestion
	    return word
```

File: tenets/codelingo/go/ignore-goroutine-return-values/example.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	"time"
+	"fmt"
+)
+
+func main() {
+	go A() // Issue
+        go B() // Non Issue
+	<-time.After(time.Second)
+
+	go func() string {
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/ignore-goroutine-return-values/example.go]:
```suggestion
	go func() string { // Issue
```

File: tenets/codelingo/go/ignore-goroutine-return-values/example.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	    word := "Hello"
+	    return word // Issue
+        }()
+
+
+
+
+        go func() {
+            fmt.Println("Hello") // Non Issue
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/ignore-goroutine-return-values/example.go]:
```suggestion
            fmt.Println("Hello")
```

File: tenets/codelingo/go/ignore-goroutine-return-values/example.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+	go func() string {
+	    word := "Hello"
+	    return word // Issue
+        }()
+
+
+
+
+        go func() {
```
Comment (BlakeMScurr) [inline tenets/codelingo/go/ignore-goroutine-return-values/example.go]:
```suggestion
        go func() { // Issue
```

File: tenets/codelingo/go/ignore-goroutine-return-values/example.go (hunk 3 above)
Comment (BlakeMScurr) [reply tenets/codelingo/go/ignore-goroutine-return-values/example.go]: Issue comments should go on the lines that'll actually be returned.

File: tenets/codelingo/go/ignore-goroutine-return-values/example.go (hunk 5 above)
Comment (CallanTaylor) [reply tenets/codelingo/go/ignore-goroutine-return-values/example.go]: I think you meant `Non Issue` here to replace the one you suggested removing in the above comment

**Pull request merged**